### PR TITLE
[css-8220] COS lite integration

### PR DESCRIPTION
**Pull request by onurmus**
* Added integration with COS lite. 
* Needed to modify dependency for worker.
* Files under lib directory is created by charmcraft, you can ignore them.
* A few screenshots from Grafana dashboard:

1.RPC Overview
image

2. Workflows 
image

3. Workflows Task Processing
image

4. Activities
image

5. Activity Task Processing
image

6. Sticky Cache
image

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -50,3 +50,14 @@ resources:
...
+provides:
+  metrics-endpoint:
+    interface: prometheus_scrape
+  grafana-dashboard:
+    interface: grafana_dashboard
+
+requires:
+  log-proxy:
+    interface: loki_push_api
+    limit: 1
```
Comment (AmberCharitos) [inline metadata.yaml:63]: Missing EOF

File: src/charm.py [hunk 2]
```diff
@@ -54,6 +59,22 @@ def __init__(self, *args):
         self.framework.observe(self.on.restart_action, self._on_restart)
         self.framework.observe(self.on.update_status, self._on_update_status)
 
+
+        # Prometheus
+        self._prometheus_scraping = MetricsEndpointProvider(
+            self,
+            relation_name="metrics-endpoint",
+            jobs=[{"static_configs": [{"targets": [f"*:9000"]}]}],
```
Comment (AmberCharitos) [inline src/charm.py:67]: I would also add this port as a literal that is imported.

File: src/resources/worker.py [hunk 3]
```diff
@@ -144,8 +154,10 @@ async def run_worker(unpacked_file_name, module_name):
             )
 
             worker_opt = WorkerOptions(sentry=sentry)
+        
+        runtime = _init_runtime_with_prometheus(9000)
```
Comment (AmberCharitos) [inline src/resources/worker.py:158]: Same comment about this port being a literal

File: src/grafana_dashboards/temporal-monitoring.json.tmpl [hunk 4]
```diff
@@ -0,0 +1,2942 @@
...
+      "1d"
+    ]
+  },
+  "timezone": "",
+  "title": "Temporal Worker SDK Metrics",
+  "variables": {
+    "list": []
+  },
+  "version": 2
+}
```
Comment (AmberCharitos) [inline src/grafana_dashboards/temporal-monitoring.json.tmpl:2942]: EOF

File: src/grafana_dashboards/temporal-monitoring.json.tmpl [hunk 5]
```diff
@@ -0,0 +1,2942 @@
+{
```
Comment (AmberCharitos) [inline src/grafana_dashboards/temporal-monitoring.json.tmpl:1]: Do you have any screenshots of the dashboard output? It's hard to review purely in this format. Or is it the same as what we have for the server?

File: lib/charms/observability_libs/v0/juju_topology.py [hunk 6]
```diff
@@ -0,0 +1,301 @@
+# Copyright 2022 Canonical Ltd.
```
Comment (AmberCharitos) [inline lib/charms/observability_libs/v0/juju_topology.py:1]: Why do you need this lib? I don't see it imported anywhere but could have just missed it.

**Review by AmberCharitos — APPROVED** [review]
LGTM - some small nit picks, great work!

File: lib/charms/observability_libs/v0/juju_topology.py (hunk 6 above)
Comment (kelkawi-a) [reply lib/charms/observability_libs/v0/juju_topology.py:1]: Looks like a transitive dependency used by Loki and Prometheus libraries.

Comment (onurmus) [reply lib/charms/observability_libs/v0/juju_topology.py:1]: Yeap, that is the reason.

File: src/grafana_dashboards/temporal-monitoring.json.tmpl (hunk 5 above)
Comment (onurmus) [reply src/grafana_dashboards/temporal-monitoring.json.tmpl:1]: Nope this is SDK-general dsahboard different from the server dashboard. Attaching some screenshots to PR definition.

File: src/literals.py [hunk 7]
```diff
@@ -21,3 +23,4 @@
     "oidc-client-cert-url",
 ]
 SUPPORTED_AUTH_PROVIDERS = ["candid", "google"]
+PROMETHEUS_PORT = 9000
```
[the remark this answers is not in the document]
Comment (kelkawi-a) [reply src/literals.py:26]: Charms can be opinionated to an extent, so I think it's fine to keep this a hardcoded value.

**Review by kelkawi-a — APPROVED** [review]
LGTM, thanks for working on this!

**Pull request merged**